File: app/src/main/java/com/scatl/uestcbbs/module/darkroom/model/DarkRoomModel.java
package com.scatl.uestcbbs.module.darkroom.model;

import com.scatl.uestcbbs.helper.rxhelper.Observer;
import com.scatl.uestcbbs.util.RetrofitCookieUtil;

import io.reactivex.Observable;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * author: sca_tl
 * date: 2021/4/10 12:09
 * description:
 */
public class DarkRoomModel {
    public void getDarkRoomList(Observer<String> observer) {
        Observable<String> observable = RetrofitCookieUtil
                .getInstance()
                .getApiService()
                .getDarkRoomList();
        observable
                .subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .subscribe(observer);

    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/darkroom/model/DarkRoomModel.kt
package com.scatl.uestcbbs.module.darkroom.model

import com.scatl.uestcbbs.util.RetrofitUtil
import io.reactivex.Observable

/**
 * Created by sca_tl at 2023/6/6 15:33
 */
class DarkRoomModel {
    fun getDarkRoomList(): Observable<String> {
        return RetrofitUtil
            .getInstance()
            .apiService
            .darkRoomList
    }

}